server/handlers/project_command_output_handler.go

```diff
@@ -85,47 +94,58 @@ func (p *DefaultProjectCommandOutputHandler) Clear(ctx models.ProjectCommandCont
 }
 
 func (p *DefaultProjectCommandOutputHandler) clearLogLines(pull string) {
-	p.controllerBufferLock.Lock()
+	p.projectOutputBuffersLock.Lock()
 	delete(p.projectOutputBuffers, pull)
-	p.controllerBufferLock.Unlock()
+	p.projectOutputBuffersLock.Unlock()
 }
 
-func (p *DefaultProjectCommandOutputHandler) addChan(pull string) chan string {
-	ch := make(chan string, 1000)
-	p.controllerBufferLock.Lock()
-	for _, line := range p.projectOutputBuffers[pull] {
-		ch <- line
-	}
+func (p *DefaultProjectCommandOutputHandler) addChan(ch chan string, pull string) {
+	p.receiverBuffersLock.Lock()
 	if p.receiverBuffers[pull] == nil {
 		p.receiverBuffers[pull] = map[chan string]bool{}
 	}
 	p.receiverBuffers[pull][ch] = true
-	p.controllerBufferLock.Unlock()
-	return ch
+	p.receiverBuffersLock.Unlock()
+
+	p.projectOutputBuffersLock.RLock()
+	for _, line := range p.projectOutputBuffers[pull] {
+		ch <- line
+	}
+	p.projectOutputBuffersLock.RUnlock()
 }
 
 //Add log line to buffer and send to all current channels
 func (p *DefaultProjectCommandOutputHandler) writeLogLine(pull string, line string) {
-	p.controllerBufferLock.Lock()
-
+	p.receiverBuffersLock.Lock()
 	for ch := range p.receiverBuffers[pull] {
 		select {
 		case ch <- line:
 		default:
+			// Buffered chan of size 10. Delete chan if still blocking
```

Comment:
Does this imply that if the channel is blocked, the ws connection is closed?  I thought we already gracefully remove it? if so is this still necessary and what other cases would hit this?

Reply:
Yes, there could be two scenarios: 
1. The client closes the ws connection gracefully, then we remove the channel in the closeHandler() and clean up resources related to that client. 
2. The client does not close the ws connection and the closeHandler() is not executed, then the channel will be blocking for n number of messages (which is the channel size) before the default block is executed which deletes the channel and cleans up the resources.

Reply:
Sounds like a good comment to add!

Reply:
Okay!